packages/headless/src/controllers/core/tab-manager/headless-core-tab-manager.ts

```diff
@@ -0,0 +1,60 @@
+import {configuration} from '../../../app/common-reducers';
+import {CoreEngine} from '../../../app/engine';
+import {tabSetReducer as tabSet} from '../../../features/tab-set/tab-set-slice';
+import {ConfigurationSection, TabSection} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+export interface TabManager extends Controller {
+  /**
+   * The state of the `TabManager` controller.
+   */
+  state: TabManagerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `TabManager` controller.
+ */
+export interface TabManagerState {
+  activeTab: string;
+}
+
+/**
+ * Creates a `TabManager` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `TabManager` controller instance.
+ */
+export function buildCoreTabManager(engine: CoreEngine): TabManager {
+  if (!loadTabReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+
+  const getCurrentTab = () => {
```

Comment:
Not super easy to read, and could be replaced by a redux selector to benefit from memoization! 😄 

```typescript
  const currentTab = createSelector(
    (state: typeof engine.state) => state.tabSet,
    (state) => {
      const activeTab = Object.values(state).find((tab) => tab.isActive);
      return activeTab?.id ?? '';
    }
  );
```

I don't think you need to fallback to `{}` if `engine.state.tabSet` is undefined, because adding the `tabSet` reducer to the engine (which you do in `loadTabReducers` implies that `tabSet` is going to be present in the engine's state, and that it's going to be set to the initial state (i.e., `{}`) if it's not already present.